python/lsst/ts/aos/analysis/stateUtils.py

```diff
@@ -0,0 +1,778 @@
+from functools import lru_cache
+from typing import Any
+
+import astropy.units as u
+import numpy as np
+import pandas as pd
+from astropy.table import QTable
+from astropy.time import Time
+from lsst.daf.butler import Butler
+from lsst.daf.butler.dimensions import DimensionRecord
+from lsst.summit.utils.efdUtils import getEfdData, getMostRecentRowWithDataBefore
+from lsst.ts.ofc import BendModeToForce, OFCData
+from lsst_efd_client import EfdClient
+
+__all__ = ["StateFetcher"]
+
+
+def m2_force_to_bending_mode(force: np.ndarray) -> np.ndarray:
+    """Convert M2 forces to bending modes
+
+    Parameters
+    ----------
+    force : `np.ndarray`
+        The M2 forces in Newtons.  The shape must be (72,)
+
+    Returns
+    -------
+    bending_mode : `np.ndarray`
+        The M2 bending modes in microns.  The shape is (20,)
+    """
+    m2_bmf = BendModeToForce("M2", OFCData("lsst"))
+    return m2_bmf.bending_mode(force)
+
+
+def m1m3_force_to_bending_mode(force: np.ndarray) -> np.ndarray:
+    """Convert M1M3 forces to bending modes
+
+    Parameters
+    ----------
+    force : `np.ndarray`
+        The M1M3 forces in Newtons.  The shape must be (156,)
+
+    Returns
+    -------
+    bending_mode : `np.ndarray`
+        The M1M3 bending modes in microns.  The shape is (20,)
+    """
+    m1m3_bmf = BendModeToForce("M1M3", OFCData("lsst"))
+    return m1m3_bmf.bending_mode(force)
+
+
+class StateFetcher:
+    def __init__(
+        self, butler: Butler, efdClient: EfdClient, instrument: str = "LSSTComCam"
```

Comment:
Include more info on class here in the init docstring as well as parameter info.